backend/store/error_groups_test.go

```diff
@@ -2,18 +2,179 @@ package store
 
 import (
 	"context"
+	"strconv"
 	"testing"
 	"time"
 
+	"github.com/aws/smithy-go/ptr"
 	"github.com/highlight-run/highlight/backend/model"
 	privateModel "github.com/highlight-run/highlight/backend/private-graph/graph/model"
 	"github.com/highlight-run/highlight/backend/util"
+	"github.com/samber/lo"
 	"github.com/stretchr/testify/assert"
 )
 
+func TestListErrorObjects(t *testing.T) {
+	util.RunTestWithDBWipe(t, "TestListErrorObjects", store.db, func(t *testing.T) {
+		errorGroup := model.ErrorGroup{
+			State: privateModel.ErrorStateOpen,
+			Event: "something broke!",
+		}
+		store.db.Create(&errorGroup)
+
+		// No error objects
+		connection, err := store.ListErrorObjects(errorGroup, ListErrorObjectsParams{})
+		assert.NoError(t, err)
+
+		assert.Equal(t, privateModel.ErrorObjectConnection{}, connection)
+
+		// One error object
+		userProperties := map[string]string{
+			"email": "chilly@mcwilly.com",
+		}
+		session := model.Session{
+			AppVersion: ptr.String("123"),
+		}
+		err = session.SetUserProperties(userProperties)
+		assert.NoError(t, err)
+
+		store.db.Create(&session)
+
+		errorObject := model.ErrorObject{
+			ErrorGroupID: errorGroup.ID,
+			SessionID:    &session.ID,
+		}
+		store.db.Create(&errorObject)
+		connection, err = store.ListErrorObjects(errorGroup, ListErrorObjectsParams{})
+		assert.NoError(t, err)
+
+		assert.Len(t, connection.Edges, 1)
+
+		edge := connection.Edges[0]
+
+		assert.Equal(t, strconv.Itoa(errorObject.ID), edge.Cursor)
+		assert.Equal(t, errorObject.ID, edge.Node.ID)
+		assert.WithinDuration(t, errorObject.CreatedAt, edge.Node.CreatedAt, 10*time.Second)
```

Comment:
Would have liked to have done a [direct equality check](https://github.com/highlight/highlight/pull/5557/commits/3cbc5cbc62d5057d1de5d394e0c09bbedd4f59d4) on the output but the time check results in a flakey test.

https://github.com/highlight/highlight/actions/runs/5203795531/jobs/9387189549?pr=5557